modules/clickhouse/main.tf

```diff
@@ -0,0 +1,43 @@
+locals {
+  dns_name_suffix = "privatelink.azure.clickhouse.cloud"
+}
+
+resource "azurerm_private_endpoint" "clickhouse" {
+  name                = "${var.namespace}-clickhouse-pe"
```

Comment:
It might be good to limit this, but in practive might not be an issue.

Azure Private Endpoint names must be between 2 and 64 characters long and meet the following requirements: 
* Start with alphanumeric characters 
* End with alphanumeric characters or an underscore 
* Contain only alphanumeric characters, underscores, periods, and hyphens 
* Be unique within a resource group

https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules

Reply:
Thanks, that is good to know, looks like this will work given our naming but I will add some validation or a note in a subsequent PR which adds more automation in this area